packages/network-controller/src/rpc-service/rpc-service.test.ts

```diff
@@ -400,14 +398,190 @@ describe('RpcService', () => {
         const promise = service.request({
           id: 1,
           jsonrpc: '2.0',
-          method: 'eth_unknownMethod',
+          method: 'eth_chainId',
           params: [],
         });
         await ignoreRejection(promise);
         expect(onBreakListener).not.toHaveBeenCalled();
       });
     });
 
+    describe.each([402, 404])(
+      'if the endpoint has a %d response',
+      (httpStatus) => {
+        it('throws a resource unavailable error without retrying the request', async () => {
+          const endpointUrl = 'https://rpc.example.chain';
+          nock(endpointUrl)
+            .post('/', {
+              id: 1,
+              jsonrpc: '2.0',
+              method: 'eth_unknownMethod',
+              params: [],
+            })
+            .reply(httpStatus);
+          const service = new RpcService({
+            fetch,
+            btoa,
+            endpointUrl,
+          });
+
+          const promise = service.request({
+            id: 1,
+            jsonrpc: '2.0',
+            method: 'eth_unknownMethod',
+            params: [],
+          });
+          await expect(promise).rejects.toThrow(
+            `RPC endpoint server error (HTTP ${httpStatus})`,
```

Comment:
I guess I wasn't doing this before, but maybe it's worth verifying that the code is correct in addition to the message? (The same goes for all of the tests in this file which verify the error message in some way.) I am just thinking that we changed for instance the code of the "invalid response" error, but that wasn't reflected in tests anywhere.